publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -154,11 +155,52 @@ public ResponseEntity send(
         }
         synchronized (this) {
             SendResult result = messageService.send(body, msgService, userDomain, tag, routingKey);
-            log.info("HTTP Status: {}",  messageService.getHttpStatus().value());
+            log.info("HTTP Status: {}", messageService.getHttpStatus().value());
             return new ResponseEntity(result, messageService.getHttpStatus());
         }
     }
 
+    /**
+     * This controller used as producer to send messages or event
+     * @param msgProtocol
+     *            message protocol (required)
+     * @param userDomain
+     *            user domain (required)
+     * @param tag
+     *            (not required)
+     * @param routingKey
+     *            (not required)
+     * @param body
+     *            (Here json body of string type as input because just to parse
+     *            the string in to JsonElement not using JsonElement directly here.)
+     * @return A response entity which contains http status and result
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @ApiOperation(value = "To publish eiffel event to message bus", response = String.class)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "Event sent successfully"),
+            @ApiResponse(code = 400, message = "Invalid event content"),
+            @ApiResponse(code = 404, message = "RabbitMq properties not found"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 503, message = "Service Unavailable")})
+    @RequestMapping(value = "/producer/msg", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity send(
+            @ApiParam(value = "message protocol", required = true) @RequestParam(value = "mp") final String msgProtocol,
+            @ApiParam(value = "user domain") @RequestParam(value = "ud", required = false) final String userDomain,
+            @ApiParam(value = "tag") @RequestParam(value = "tag", required = false) final String tag,
+            @ApiParam(value = "routing key") @RequestParam(value = "rk", required = false) final String routingKey,
+            @ApiParam(value = "eiffel event", required = true) @RequestBody final String body) {
+        try {
+            JsonElement inputBody = JsonParser.parseString(body);
+            return send(msgProtocol, userDomain, tag, routingKey, inputBody);
+        } catch (JsonSyntaxException e) {
+            String exceptionMessage = e.getMessage();
+            log.error("Unexpected exception caught due to parsed json data", exceptionMessage);
```

Comment:
Done

Reply:
If we use inputBodyhere in the catch section we need initiailize this and it seems null in the catch, so it is not better to use **body** instead of **inputBody**?